File: spec/factories/profiles.rb
# == Schema Information
#
# Table name: profiles
#
#  id           :integer          not null, primary key
#  user_id      :integer
#  name         :string
#  kana         :string
#  organization :string
#  introduction :text
#  image        :binary
#  created_at   :datetime         not null
#  updated_at   :datetime         not null
#

FactoryBot.define do
  factory :profile do
    name         { Faker::Name.name }
    kana         { Faker::Name.name }
    organization { Faker::Company.name }
    introduction { Faker::Lorem.characters(80) }
    image        { fixture_file_upload(Rails.root.join('spec', 'images', 'profile', 'default.jpg'), 'image/jpg') }
  end
end

File: frontend/src/api/__test__/factories/setting/profiles.js
var faker = require('faker')
faker.locale = 'ja'

export default {
  Profile: {
    profile1: {
      id: 1,
      user_id: 1,
      name:         faker.name.firstName() + faker.name.lastName(),
      kana:         faker.name.firstName() + faker.name.lastName(),
      organization: faker.lorem.text(),
      introduction: faker.lorem.text(),
      image:        faker.image.people(),
      created_at:   faker.date.recent(),
      updated_at:   faker.date.recent()
    },
    profile2: {
      id: 2,
      user_id: 2,
      name:         faker.name.firstName() + faker.name.lastName(),
      kana:         faker.name.firstName() + faker.name.lastName(),
      organization: faker.lorem.text(),
      introduction: faker.lorem.text(),
      image:        faker.image.people(),
      created_at:   faker.date.recent(),
      updated_at:   faker.date.recent()
    }
  }
}
